dispatch.go

```diff
@@ -89,22 +90,35 @@ func newDispatch(enc encoder, dec byteReadingDecoder, l LogInterface, wef WrapEr
 
 type call struct {
 	ch             chan error
+	doneCh         chan struct{}
 	method         string
 	seqid          int
 	arg            interface{}
 	res            interface{}
 	errorUnwrapper ErrorUnwrapper
 	profiler       Profiler
+	ctx            context.Context
```

Comment:
Cool, I'll make the call inherit `context.Context` anonymously to make it more explicit that the call _is a_ context.